retrofit/src/main/java/retrofit2/Retrofit.java

```diff
@@ -163,12 +163,10 @@ private void eagerlyValidateMethods(Class<?> service) {
     ServiceMethod<?> result = serviceMethodCache.get(method);
     if (result != null) return result;
 
-    synchronized (serviceMethodCache) {
-      result = serviceMethodCache.get(method);
-      if (result == null) {
-        result = ServiceMethod.parseAnnotations(this, method);
-        serviceMethodCache.put(method, result);
-      }
+    result = serviceMethodCache.get(method);
+    if (result == null) {
+      result = ServiceMethod.parseAnnotations(this, method);
+      serviceMethodCache.put(method, result);
```

Comment:
What we probably want here is to putIfAbsent so that if we race both callers get the same instance back.

Reply:
Yeah,the code block is same as putIfAbsent(), but this don't cause thread-safe problem,so 
 remove 'synchronized' is ok based on performance.

Reply:
It's not the same. Please update.